### ISSUE TITLE
Resource aws_ami doesn't support import

### ISSUE THREAD

**Issue by finferflu** [terraform-providers/terraform-provider-aws#4907]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AWS provider currently doesn't support importing `aws_ami` resources.

### New or Affected Resource(s)

* aws_ami (Terraform `v0.11.7`, provider.aws `v1.24.0`)

**Comment by tomelliff** [comment]
Why would you want to import an AMI for Terraform to manage? If you want to be able to refer to an AMI you should be able to use the [`aws_ami` data source](https://www.terraform.io/docs/providers/aws/d/ami.html).

The [`aws_instance` resource docs](https://www.terraform.io/docs/providers/aws/r/instance.html) give a good example of this:
```hcl
data "aws_ami" "ubuntu" {
  most_recent = true

  filter {
    name   = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  owners = ["099720109477"] # Canonical
}

resource "aws_instance" "web" {
  ami           = "${data.aws_ami.ubuntu.id}"
  instance_type = "t2.micro"

  tags {
    Name = "HelloWorld"
  }
}
```

**Comment by finferflu** [comment]
@tomelliff, thanks for getting back to me. I am using Packer to generate AMIs, which means that they stay unmanaged. I would have liked to import them via Terraform so that they could be appropriately destroyed/recreated as needed, and for that reason using a data source doesn't help. I hope that makes sense.

Thank you!

**Comment by mchouque** [comment]
I have the same exact need, as in I want to import my own image. Right now I convert them using aws ec2 import-image.

**Comment by bflad** [comment]
Pull request submitted: #5990

**Comment by mchouque** [comment]
So if my understanding is correct, this pull request would solve the initial problem but wouldn't solve the step of creating an AMI from a disk image contained in a S3 bucket, right?

**Comment by bflad** [comment]
@mchouque you are correct. Could you submit a new, separate feature request issue with details about your use case and potential configuration? I bet a fresh issue would garner more attention than this existing issue which might be slightly confusing as both Terraform and EC2 have differing meanings for "import". 😄

**Comment by mchouque** [comment]
@bflad Thanks for the feedback, I created #5998 about that.

**Comment by bflad** [comment]
@mchouque great write up! 💯

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
